### PR TITLE
fix: should not set illegal locale

### PR DESCRIPTION
**Pull request by Decodetalkers**
Issue: https://github.com/linuxdeepin/developer-center/issues/6741 Log:

### CODE REVIEW

File: src/plugin-datetime/operation/datetimeworker.h [hunk 1]
```diff
@@ -78,13 +78,15 @@ private Q_SLOTS:
...
     void initRegionFormatData();
+    std::optional<QStringList> getSupportedLocale();
 
 private:
     DatetimeModel *m_model;
     DatetimeDBusProxy *m_timedateInter;
     QDateTime *m_setDatetime;
     RegionProxy *m_regionInter;
     DTK_CORE_NAMESPACE::DConfig *m_config;
+    std::optional<QStringList> m_supportedLocale;
```
Comment (dengbo11) [inline src/plugin-datetime/operation/datetimeworker.h]: m_supportedLocaleList

File: src/plugin-datetime/operation/datetimeworker.cpp [hunk 2]
```diff
@@ -346,7 +366,21 @@ void DatetimeWorker::genLocale(const QString &localeName)
 
     QSettings settings(localeConfPath, QSettings::IniFormat);
 
-    QString localeSet = localeName + ".UTF-8";
+    auto supportedLocaleRes = getSupportedLocale();
+    if (!supportedLocaleRes.has_value()) {
+        return;
+    }
+    QStringList supportedLocale = supportedLocaleRes.value();
```
Comment (dengbo11) [inline src/plugin-datetime/operation/datetimeworker.cpp]: supportedLocaleList

**Review by wineee — DISMISSED** [review]

**Review by dengbo11 — APPROVED** [review]

**Pull request merged**